.github/workflows/ci.yml

```
@@ -0,0 +1,65 @@
+name: Quality Assurance
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build-linux:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+        platform: [x32, x64]
+        compiler: [gcc, clang]
+    steps:
+      - name: Setup Dependencies
+        run: |
+          sudo apt-get update -y -qq
+          sudo apt-get install -y gcc g++ make autoconf automake make libyajl-dev libxml2-dev libmaxminddb-dev libpcre2-dev libpcre2-8-0 libpcre2-16-0 libpcre2-32-0 libcurl4-gnutls-dev jq wget
+      - name: Install ModSecurity library
+        run: |
+          modsec=`curl -s https://api.github.com/repos/owasp-modsecurity/ModSecurity/releases/latest | jq .assets[0].browser_download_url | tr -d "\""`
+          modsecver=`curl -s https://api.github.com/repos/owasp-modsecurity/ModSecurity/releases/latest | jq .name | tr -d "\""`
+          wget $modsec -O modsecurity.tar.gz
+          tar -xzf modsecurity.tar.gz
+          cd modsecurity-$modsecver
+          ./configure --disable-lmdb --prefix=/usr
+          make -j `nproc`
+          sudo make install
+          cd
+      - uses: actions/checkout@v2
```

Use versions tagged to hashes. You can later set up Dependabot or Renovate to automatically update these dependencies in your workflows periodically.

```suggestion
      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

---

Thanks, actually I don't want to use hashes. It would be fine to align each affected repositories (ModSecurity (with both version)), but now those uses the version syntax.

I try to keep it my mind, and that will be the next step. Thanks again.

---

TBH, I don't see the advantage of aligning the app versions across repositories. I think pinning is much more important.